src/HealthChecks.Publisher.ApplicationInsights/ApplicationInsightsPublisher.cs

```diff
@@ -123,5 +123,10 @@ private TelemetryClient GetOrCreateTelemetryClient()
             }
             return _client;
         }
+
+        public void Dispose()
+        {
+            _telemetryConfiguration?.Dispose();

```

Comment:
I suggest to post another PR with changes in `ApplicationInsightsPublisher` since it has nothing to do with CloudWatch.

Reply:
Also I'm not sure that value from `IOptions<>` object **should be** disposed at all since it is shared across all program.

Reply:
I mean one should always follow golden rule to dispose only those resources that he creates himself.